components/video-hearings/subscriptions.tf

```diff
@@ -9,10 +9,19 @@ locals {
 module "subscription" {
   for_each = local.subscriptions
 
-  source = "../../modules/subscription"
-  name   = each.key
-  value  = each.value
-
+  source                  = "../../modules/subscription"
+  name                    = each.key
+  value                   = each.value
+  common_tags             = module.tags[each.key].common_tags
+  environment             = try(each.value.environment, (lower(replace([each.key][0], "/.*-([A-Za-z]+).*/", "$1"))))
   billing_account_name    = var.billing_account_name
   enrollment_account_name = "TODO"
 }
+
+module "tags" {
+  for_each    = local.subscriptions
+  source      = "git::https://github.com/hmcts/terraform-module-common-tags.git?ref=master"
+  environment = try(each.value.environment, (lower(replace([each.key][0], "/.*-([A-Za-z]+).*/", "$1"))))
```

Comment:
could the regex be extracted to a local so it's not repeated and it can be given a clear name explaining what it's doing?